test/framework/commands.go

```diff
@@ -1,13 +1,40 @@
 package framework
 
-type commandOpt func(*[]string)
+import "github.com/aws/eks-anywhere/pkg/semver"
 
-func appendOpt(new ...string) commandOpt {
-	return func(args *[]string) {
+type CommandOpt func(*string, *[]string) (err error)
+
+func appendOpt(new ...string) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
 		*args = append(*args, new...)
+		return nil
 	}
 }
 
-func withKubeconfig(kubeconfigFile string) commandOpt {
+func withKubeconfig(kubeconfigFile string) CommandOpt {
 	return appendOpt("--kubeconfig", kubeconfigFile)
 }
+
+func ExecuteWithEksaVersion(version *semver.Version) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
+		b, err := GetReleaseBinaryFromVersion(version)
+		*binaryPath = b
+		return err
+	}
+}
+
+func ExecuteWithLatestMinorReleaseFromVersion(version *semver.Version) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
+		b, err := GetLatestMinorReleaseBinaryFromVersion(version)
+		*binaryPath = b
+		return err
+	}
+}
+
+func ExecuteWithLatestMinorReleaseFromMain() CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
+		b, err := GetLatestMinorReleaseBinaryFromMain()
```

Comment:
No need to implement here, just for the future 
Since we are calling these methods from all over, should we make them a bit smart so they don't do anything if the binary is already in place? I know they won't download binaries twice, but they still download the release manifest every time. Maybe just an idea for a future fun optimization time

Reply:
I'll add it to the stories